### DISCUSSION TITLE
Max9814 to mp3 sdfat example

### DISCUSSION THREAD

**Discussion by luke2023** [pschatzmann/arduino-audio-tools#390] [Q&A] [answered]
I have made a beTA version of adc to mp3 example,but I don't have max9814 now ,so I'll simply upload my beta code
Question ; I don't know if I need to create a empty mp3

**Comment by luke2023** [comment]
V0.1

```
////input:gpio 34
#include "Arduino.h"
#include "AudioTools.h"
#include "SdFat.h"
const uint16_t sample_rate =44100;
AnalogAudioStream in; 
int channels = 1;
SdFat SD;
File audioFile;                                                   // final output stream
EncodedAudioStream out(&audioFile, new MP3EncoderLAME());             // encode as mp3 file  
StreamCopy copier(out, in); // copy i2sStream to CsvStream
// Arduino Setup
void setup(void) {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // RX automatically uses port 0 with pins GPIO34,GPIO35
  auto cfgRx = in.defaultConfig(RX_MODE);
  cfgRx.sample_rate = sample_rate;
  cfgRx.channels = channels;
  in.begin(cfgRx);

 auto cfg = noise.defaultConfig();
  cfg.sample_rate = sample_rate;
  cfg.channels = channels;
  cfg.bits_per_sample = 16;
  
  out.begin(cfg);
SD.begin(SdSpiConfig(PIN_CS, DEDICATED_SPI, SD_SCK_MHZ(2)));
  audioFile = SD.open("/test/002.mp3", O_WRITE | O_CREAT);//how to create empty mp3 with esp32?
}

// Arduino loop - copy data 
void loop() {
  copier.copy();
}


```

**Comment by pschatzmann** [comment, marked as answer]
You need to add some logic to close the file: otherwise it will be empty.
To overwrite the file you can just move to the beginning after opening it. If I remember right it's with audioFile.seek(0)
I am not sure if this logic will not work: You define what needs to be done with the file even before it (really) exists.

**Reply by luke2023** [reply]
@pschatzmann Thank you for your suggestion!And do I need to create a empty mp3 file  in my computer?And what does this "o_creat"and "o_write"means?

audioFile = SD.open("/test/002.mp3", O_WRITE | O_CREAT);

**Comment by luke2023** [comment]
@pschatzmann  Thank you for your suggestion!And do I need to create a empty mp3 file in my computer?And what does this "o_creat"and "o_write"means?

audioFile = SD.open("/test/002.mp3", O_WRITE | O_CREAT);

**Comment by pschatzmann** [comment]
Google is your friend: this has nothing to do with my library!

**Comment by luke2023** [comment]
@pschatzmann hi,is this normal?thank you!![image](https://user-images.githubusercontent.com/110216618/194583724-ae39fcc5-a463-4072-936c-ca8cf6a65ecb.jpeg)
